x/concentrated-liquidity/incentives_test.go

```diff
@@ -593,12 +599,75 @@ func (s *KeeperTestSuite) TestUpdateUptimeAccumulatorsToNow() {
 
 				// Ensure that LastLiquidityUpdate field is updated for pool
 				s.Require().Equal(s.Ctx.BlockTime(), clPool.GetLastLiquidityUpdate())
-
 				// Ensure that pool's IncentiveRecords are updated to reflect emitted incentives
-				s.Require().Equal(tc.expectedIncentiveRecords, clPool.GetPoolIncentives())
+				updatedIncentiveRecords, err := clKeeper.GetAllIncentiveRecordsForPool(s.Ctx, tc.poolId)
+				s.Require().NoError(err)
+				s.Require().Equal(tc.expectedIncentiveRecords, updatedIncentiveRecords)
 			} else {
 				s.Require().Error(err)
 			}
 		})
 	}
 }
+
+// Note: we test that incentive records are properly deducted by emissions in `TestUpdateUptimeAccumulatorsToNow` above.
+// This test aims to cover the behavior of a series of state read/writes relating to incentive records.
+func (s *KeeperTestSuite) TestIncentiveRecordsSetAndGet() {
+	s.SetupTest()
+	clKeeper := s.App.ConcentratedLiquidityKeeper
+	s.Ctx = s.Ctx.WithBlockTime(defaultStartTime)
+	emptyIncentiveRecords := []types.IncentiveRecord{}
+
+	// Set up test pool
+	clPoolOne := s.PrepareConcentratedPool()
+
+	// Set up second pool for reference
+	clPoolTwo := s.PrepareConcentratedPool()
+
+	// Ensure both pools start with no incentive records
+	poolOneRecords, err := clKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolOne.GetId())
+	s.Require().NoError(err)
+	s.Require().Equal(emptyIncentiveRecords, poolOneRecords)
+
+	poolTwoRecords, err := clKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolTwo.GetId())
+	s.Require().NoError(err)
+	s.Require().Equal(emptyIncentiveRecords, poolTwoRecords)
+
+	// Ensure setting and getting a single record works with single Get and GetAll
+	clKeeper.SetIncentiveRecord(s.Ctx, incentiveRecordOne)
+	poolOneRecord, err := clKeeper.GetIncentiveRecord(s.Ctx, clPoolOne.GetId(), incentiveRecordOne.IncentiveDenom, incentiveRecordOne.MinUptime)
+	s.Require().NoError(err)
+	s.Require().Equal(incentiveRecordOne, poolOneRecord)
+	allRecordsPoolOne, err := clKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolOne.GetId())
+	s.Require().NoError(err)
+	s.Require().Equal([]types.IncentiveRecord{incentiveRecordOne}, allRecordsPoolOne)
+
+	// Ensure records for other pool remain unchanged
+	_, err = clKeeper.GetIncentiveRecord(s.Ctx, clPoolTwo.GetId(), incentiveRecordOne.IncentiveDenom, incentiveRecordOne.MinUptime)
```

Comment:
nit: can we check the output of this is nil as well

Reply:
Done!